### PR TITLE
Sam changes

### PR DESCRIPTION
**Pull request by sbarguil**
Innitial proposal of the **vpn-types.yang**  it includes:

Encapsulation Type
Tag-Type
Underlay Transport
Service Topology
Service Profiles

### CODE REVIEW

File: yang/IETF-04/ietf-vpn-types.yang [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+  identity RSVP-TE {
+    base protocol-type ;
+    description
+      "Transport based on SR.";
+  }
+
+  identity unknown {
+    base protocol-type ;
+    description
+      "Transport based on SR.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:200]:
```suggestion
      "Not known at this stage.";
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+identity protocol-type {
+      description
+      "Base identity for Protocol Type.";
+  }
+
+  identity GRE {
+    base protocol-type ;
+    description
+      "Transport based on GRE.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:164]:
```suggestion
         "GRE encapsulation.";
       reference
         "RFC 1701: Generic Routing Encapsulation (GRE)
          RFC 1702: Generic Routing Encapsulation over IPv4 networks
          RFC 7676: IPv6 Support for Generic Routing Encapsulation
                    (GRE)";
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    description
+      "Using both a CVLAN tag and an SVLAN tag.";
+  }
+
+identity protocol-type {
+      description
+      "Base identity for Protocol Type.";
+  }
+
+  identity GRE {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:161]:
```suggestion
  identity gre {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on GRE.";
+  }
+
+  identity LDP {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:167]:
```suggestion
  identity ldp {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on LDP.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:170]: We need to provide a description of what is meant by transport on LDP. Please add a reference clause.

File: yang/IETF-04/ietf-vpn-types.yang [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+  }
+  
+  identity BGP {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:173]:
```suggestion
  identity bgp {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on BGP.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:176]: Idem as LDP

File: yang/IETF-04/ietf-vpn-types.yang [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+  }
+
+  identity SR {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:179]:
```suggestion
  identity sr {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on SR.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:182]: Idem as LDP

File: yang/IETF-04/ietf-vpn-types.yang [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+  }
+
+  identity SR-TE {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:185]:
```suggestion
  identity sr-te {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 11]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on SR.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:188]: Please update the description to reflect the label

File: yang/IETF-04/ietf-vpn-types.yang [hunk 12]
```diff
@@ -0,0 +1,346 @@
...
+  }
+
+  identity RSVP-TE {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:191]:
```suggestion
  identity rsvp-te {
```

File: yang/IETF-04/ietf-vpn-types.yang [hunk 13]
```diff
@@ -0,0 +1,346 @@
...
+    base protocol-type ;
+    description
+      "Transport based on SR.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-types.yang:194]: Update the description to reflect the label.

File: yang/IETF-04/ietf-vpn-common.yang [hunk 14]
```diff
@@ -0,0 +1,505 @@
...
+  "Allows the VPN to connect to a CSP.";
+ }
+
+/* Typedef */
+
+/* Identities */
+
+  identity operational-status {
+        description
+        "Base identity for Protocol Type.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:77]:
```suggestion
        "Base identity for the operational status.";
```

File: yang/IETF-04/ietf-vpn-common.yang [hunk 15]
```diff
@@ -0,0 +1,505 @@
...
+  identity operational-state-unknown {
+      base operational-status;
+          description
+            "Operational status UNKNOWN.";
+        }
+
+
+identity administrative-status {
+        description
+        "Base identity for Protocol Type.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:101]:
```suggestion
        "Base identity for administrative status.";
```

File: yang/IETF-04/ietf-vpn-common.yang [hunk 16]
```diff
@@ -0,0 +1,505 @@
...
+  identity administrative-state-down {
+      base administrative-status;
+          description
+            "Administrative status DOWN/Disabled.";
+        }
+
+  identity administrative-state-testing {
+      base administrative-status;
+          description
+            "Administrative status UNKNOWN.";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:119]:
```suggestion
            "Administrative status is up for testing purposes.";
```

File: yang/IETF-04/ietf-vpn-common.yang [hunk 17]
```diff
@@ -0,0 +1,505 @@
...
+        }
+
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:121]:
```suggestion
  identity administrative-state-pre-deployment {
      base administrative-status;
      description
          "Administrative status pre-deployment phase.";
   }
  
```

File: yang/IETF-04/ietf-vpn-common.yang [hunk 18]
```diff
@@ -0,0 +1,505 @@
...
+         stopped (DOWN).";
+    }
+    description
+      "This grouping defines some operational
+       parameters for the service.";
+  }
+
+  grouping service-status {
+    container status {
+      leaf admin-status {
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:385]: I thought we agreed to have the timestamp for admin-status, but as an optional attribute.

File: yang/IETF-04/ietf-vpn-common.yang [hunk 19]
```diff
@@ -0,0 +1,505 @@
...
+  revision 2020-06-03 {
+    description
+      "Initial revision.";
+    reference
+      "RFC XXXX: A VPN common types YANG Model";
+      // RFC Ed.: replace XXXX with actual RFC number and remove
+     // this note
+  }
+
+/* Feature */
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:64]: Shouldn't we include "common" features between L3SM/L3SN as well?

Comment (sbarguil) [reply yang/IETF-04/ietf-vpn-common.yang:64]: Yes, I will start with it.

Comment (sbarguil) [reply yang/IETF-04/ietf-vpn-common.yang:64]: My key question is @oscargdd @boucadair may I upload the L2NM in this repo?

File: yang/IETF-04/ietf-vpn-common.yang [hunk 20]
```diff
@@ -0,0 +1,510 @@
...
+
+     This version of this YANG module is part of RFC XXXX
+     (https://www.rfc-editor.org/info/rfcXXXX); see the RFC itself
+     for full legal notices.";
+
+  revision 2020-06-03 {
+    description
+      "Initial revision.";
+    reference
+      "RFC XXXX: A VPN common types YANG Model";
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:59]:
```suggestion
      "RFC XXXX: A Layer 3 VPN Network YANG Model";
```

This is the name of the future RFC, not the one of the module.

File: yang/IETF-04/ietf-vpn-common.yang [hunk 21]
```diff
@@ -0,0 +1,510 @@
...
+                  <mailto:oscar.gonzalezdedios@telefonica.com>
+        Author:   Mohamed Boucadair
+                  <mailto:mohamed.boucadair@orange.com>           
+        Author:    Luis Angel Munoz
+                  <mailto:luis-angel.munoz@vodafone.com>
+        Author:    Alejandro Aguado
+                  <mailto:alejandro.aguado_martin@nokia.com> 
+    ";
+  description
+    "This YANG module defines a generic network-oriented model
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:39]:
```suggestion
    "This YANG module defines a common module that is meant
    to be reused by various VPN-related modules (e.g., L3NM, L2NM).
```

File: yang/IETF-04/ietf-vpn-common.yang [hunk 22]
```diff
@@ -0,0 +1,510 @@
...
+     for the configuration of Layer 3 Virtual Private Networks.
```
Comment (boucadair) [inline yang/IETF-04/ietf-vpn-common.yang:40]:
```suggestion
```

**Pull request merged**